### PR TITLE
Add some new translations for de dictionary

### PR DESCRIPTION
**Pull request by milespop**
#### Description of changes
Adding new keys to the German translations for

1. Sign Up with Facebook
2. Sign Up with Google
3. Enter your Password
4. Enter your Email

#### Description of how you validated changes

Imported the project to my own with `yarn link` and checked the translations:
image
image

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated - N/A
- [X] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [X] Relevant documentation is changed or added (and PR referenced)  - N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/i18n/dictionaries/authenticator/de.ts [hunk 1]
```diff
@@ -66,6 +66,10 @@ export const deDict: AuthenticatorDictionary = {
   'Confirm Sign In': 'Anmeldung bestätigen',
   'Create account': 'Hier registrieren',
   'Enter your password': 'Geben Sie Ihr Passwort ein',
+  'Enter your Password': 'Geben Sie Ihre Passwort',
```
Comment (reesscot) [inline packages/ui/src/i18n/dictionaries/authenticator/de.ts:69]: Do we need both of these? Shouldn't line 69 replace 68?

Comment (calebpollman) [inline packages/ui/src/i18n/dictionaries/authenticator/de.ts:69]: Should the corresponding value of `'Enter your Password'` align with the value of `'Enter your password'` (or vice versa)?

Comment (milespop) [reply packages/ui/src/i18n/dictionaries/authenticator/de.ts:69]: Updated

**Comment by milespop** [comment]
Hi  @ericclemmons and @calebpollman - I've left matching entries to ensure this wouldn't break existing translations. Would it be beneficial to update the placeholders for consistency then we wouldn't need the almost duplicate entries.

I'm on version @aws-amplify/ui-react@4.3.3 and the placeholders have the capitalization in Password/Email

**Review by calebpollman — DISMISSED** [review]
🚢

**Review by wlee221 — APPROVED** [review]
Thanks for your contribution!

**Review by calebpollman — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**